### PR TITLE
overlap scrolling

### PR DESCRIPTION
**Pull request by linuxCowboy**
let a few lines overlap

### CODE REVIEW

File: src/chmhtmlwindow.cpp
Comment (rzvncj) [inline src/chmhtmlwindow.cpp:359]: But this makes `WXK_PAGEDOWN` behave differently than `WXK_SPACE`, right?

**Review by rzvncj — CHANGES_REQUESTED** [review]
This pull request appears to make the SPACE key behave differently than the Page Down key. It also appears to only affect page down behaviour, but not page up.

File: src/chmhtmlwindow.cpp
Comment (linuxCowboy) [reply src/chmhtmlwindow.cpp:359]: Yes, intended. 

The Page Up overlap is not relevant and I wanted to create the least intrusive patch.

Comment (rzvncj) [reply src/chmhtmlwindow.cpp:359]: I've signaled two issues here. They are both relevant.

1. With your patch, if you press the Space key, xCHM does something (page down with some of the previous text kept on-screen). If you press the Page Down key, it does _something else_. That is not OK.

2. Assuming we're after keeping some of the previous text on-screen for Page {Up, Down}, Page Up behaves differently than Page Down. That is also not OK.

"Least intrusiveness" is not in itself a goal dictating patch acceptance. It is, of course, desirable if possible, but the important thing is that the patches are correct and complete in fixing what they purport to fix.

https://www.goodreads.com/quotes/7133605-make-things-as-simple-as-possible-but-no-simpler

Comment (linuxCowboy) [reply src/chmhtmlwindow.cpp:359]: It goes without saying for me, that some lines _should_ overlap when reading.

The flaw lies in the **wxwidgets**. I needed a quick fix to read an old chm-book.

Change or discard the patch. Afterall, it's your program.

Thank you for sharing!

Comment (rzvncj) [reply src/chmhtmlwindow.cpp:359]: You're missing my point. I'm not saying that text should not overlap. I'm also not saying that I'm not interested in your contribution (in fact I do appreciate your pull request).

I _am_ saying this:

- If you're adding this, it needs to be added the right way. That means that Space and Page Down need to do the same thing, and that Page Up should also keep some of the previous text (the same amount, in fact) on the page. I won't just take the patch without consistent (and well-tested) behaviour.
- If you think that the root problem is wxWidgets, you should fix wxWidgets (and open up a pull request on their project, thus solving the problem for _all_ software projects that use it).
- If there's a feature you want, and you're contributing to a free software project, you should add a fully fledged version of the changes you want, or at least be willing to take direction from the software maintainer(s) until your pull request meets the merge criteria.

I don't get paid for xCHM, not having text left over from the previous screen does not bother me, and you're the only user complaining about this, so this is a problem you want fixed for yourself, and the great thing about [FLOSS](https://en.wikipedia.org/wiki/Free_and_open-source_software) is that it lets _you_ change it to suit your needs. So in way it's not _mine_, but _ours_, but _ours_ means we get to share both the result _and_ the work.

Comment (linuxCowboy) [reply src/chmhtmlwindow.cpp:359]: I see no reasonable reason why two different keys have to do the same.

Anyway, here's a patch that should match your criteria.

Comment (rzvncj) [reply src/chmhtmlwindow.cpp:359]: All reasons are reasonable by definition. The two different keys should _continue_ to be tied to the same action. The reason should be clear from this:
```c++
352     case WXK_SPACE:
353         event.m_keyCode = WXK_PAGEDOWN;
354         break;
```
which is the original code, which effectively tells wxWidgets "when the user presses the Space key, do whatever you were doing for the Page Down key". You're changing application behaviour and expect me to provide a reason why you shouldn't, when in fact it is you that bear the burden of proof for the change.

Comment (rzvncj) [reply src/chmhtmlwindow.cpp:359]: I've added https://github.com/rzvncj/xCHM/commit/cc5597cec751a9cae59a1222f3e01b453405c9b3.

**Pull request closed without merging**
